### ISSUE TITLE
Boogie error: Only identifiers are supported as function calls

### ISSUE THREAD

**Issue by dddejan** [SRI-CSL/solidity#14]
Error:

```
./solc-verify.py test/libsolidity/syntaxTests/nameAndTypeResolution/255_using_for_function_on_struct.sol --output .
...
======= test/libsolidity/syntaxTests/nameAndTypeResolution/255_using_for_function_on_struct.sol =======
test/libsolidity/syntaxTests/nameAndTypeResolution/255_using_for_function_on_struct.sol:6:16: Boogie error: Only identifiers are supported as function calls
        return x.mul(a);
               ^------^
...
```

**Comment by hajduakos** [comment]
Fixed in 542c5e4

**Comment by dddejan** [comment]
I see in the translation that the library multiplication is translated into

```
mul#18(__this: address_t, __msg_sender: address_t, __msg_value: int, self#5: address_struct_stor_s#3, x#7: int)
```

My understanding of Solidity is that there is no need for __this and other special members. By solidity docs "t can only access state variables of the calling contract if they are explicitly supplied".
